rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -705,17 +716,27 @@ void configure_qos_for_security(
     qos, "dds.sec.access.permissions", "permissions.p7s",
     security_options);
 
-  dds_qset_prop(qos, "dds.sec.auth.library.path", "libdds_security_auth.so");
+  dds_qset_prop(qos, "dds.sec.auth.library.path", "dds_security_auth");
   dds_qset_prop(qos, "dds.sec.auth.library.init", "init_authentication");
   dds_qset_prop(qos, "dds.sec.auth.library.finalize", "finalize_authentication");
 
-  dds_qset_prop(qos, "dds.sec.crypto.library.path", "libdds_security_crypto.so");
+  dds_qset_prop(qos, "dds.sec.crypto.library.path", "dds_security_crypto");
   dds_qset_prop(qos, "dds.sec.crypto.library.init", "init_crypto");
   dds_qset_prop(qos, "dds.sec.crypto.library.finalize", "finalize_crypto");
 
-  dds_qset_prop(qos, "dds.sec.access.library.path", "libdds_security_ac.so");
+  dds_qset_prop(qos, "dds.sec.access.library.path", "dds_security_ac");
   dds_qset_prop(qos, "dds.sec.access.library.init", "init_access_control");
   dds_qset_prop(qos, "dds.sec.access.library.finalize", "finalize_access_control");
+
+  return RMW_RET_OK;
+#else
+  (void) qos;
+  (void) security_options;
+  RMW_SET_ERROR_MSG(
+    "Security was requested but the Cyclone DDS being used does not have security "
+    "support enabled. Recompile with CMake option '-DENABLE_SECURITY' Cmake option.");
```

Comment:
This is still a bit wonky:

```suggestion
    "Security was requested but the Cyclone DDS being used does not have security "
    "support enabled. Recompile Cyclone DDS with the '-DENABLE_SECURITY' CMake option.");
```

Reply:
> ``
```suggestion
    "Security was requested but the Cyclone DDS being used does not have security "
    "support enabled. Update and recompile with CMake option '-DENABLE_SECURITY' Cmake option.");
```
This may also require getting the latest build.

Reply:
Only for the very near term. Once this is released they will never need to update; I don't think it's worth specifying. It can also be harmful and confusing if they're building a release, like Eloquent, and indeed should NOT update.

Reply:
Also, I think you need to be specific: recompile CYCLONE. Otherwise it can be misunderstood as "recompile the thing giving you this error" which is actually RMW.